ceph/utils.py

```diff
@@ -165,6 +165,33 @@ def volume_cleanup(volume, osp_cred):
                     return 1
 
 
+# Added this function to cleanup stale volumes with status deleting, available, error, unknown, reserved
+def cleanup_ceph_vols(osp_cred, pattern=None, timeout=300):
+    driver = get_openstack_driver(osp_cred)
+    timeout = datetime.timedelta(seconds=timeout)
+    for volume in driver.list_volumes():
+        log.info("volume %s status is %s", volume.name, volume.state)
+    with parallel() as p:
+        for volume in driver.list_volumes():
+            if volume.state in "deleting":
+                log.info("volume %s %s status is %s", volume.id, volume.name, volume.state)
```

Comment:
this works, but IMO declaring a tuple list for more than one would be nice.

Reply:
Update to tuple